psa_car_controller/psa/constants.py

```diff
@@ -28,7 +28,7 @@
     "program3": {"day": [0, 0, 0, 0, 0, 0, 0], "hour": 34, "minute": 7, "on": 0},
     "program4": {"day": [0, 0, 0, 0, 0, 0, 0], "hour": 34, "minute": 7, "on": 0}
 }
-AUTHORIZE_SERVICE = "https://api.mpsa.com/api/connectedcar/v2/oauth/authorize"
+AUTHORIZE_SERVICE = "https://idpcvs.opel.com/am/oauth2/authorize"
```

Comment:
Has this change been tested for other brands?

Reply:
For lack of test accounts: no. Stellantis seems to be doing all kinds of crazy OAuth proxying internally so maybe if we're lucky the endpoints can just be used interchangeably. I found the original URL from the Android app's logcat output, the others I just found them in classes4.dex so I just made AUTHORIZE_SERVICE a dict too.

classes4.dex.txt:43f3a0: 1a02 bd90                              |0012: const-string v2, "https://idpcvs.driveds.com/am/oauth2/authorize" // string@90bd
classes4.dex.txt:43f3ec: 1a02 c390                              |0038: const-string v2, "https://idpcvs.peugeot.com/am/oauth2/authorize" // string@90c3
classes4.dex.txt:43f438: 1a02 ba90                              |005e: const-string v2, "https://idpcvs.citroen.com/am/oauth2/authorize" // string@90ba
classes4.dex.txt:43f488: 1a02 c090                              |0086: const-string v2, "https://idpcvs.opel.com/am/oauth2/authorize" // string@90c0
classes4.dex.txt:43f4da: 1a02 c690                              |00af: const-string v2, "https://idpcvs.vauxhall.co.uk/am/oauth2/authorize" // string@90c6

Reply:
Internal server error again !!

Traceback (most recent call last):
  File "C:\Users\vipaz\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\psa_car_controller\web\view\config_views.py", line 133, in connectPSA
    res = firstLaunchConfig(app_name, email, password, countrycode)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\vipaz\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\psa_car_controller\psa\setup\app_decoder.py", line 95, in firstLaunchConfig
    psacc.connect(client_email, client_password)
  File "C:\Users\vipaz\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\psa_car_controller\psacc\application\psa_client.py", line 35, in connect
    self.manager.init_with_user_credentials_realm(user, password, self.realm)
  File "C:\Users\vipaz\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\psa_car_controller\psa\oauth.py", line 26, in init_with_user_credentials_realm
    self._token_request(self._grant_password_request_realm(login, password, realm), True)
  File "C:\Users\vipaz\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\oauth2_client\credentials_manager.py", line 205, in _token_request
    CredentialManager._handle_bad_response(response)
  File "C:\Users\vipaz\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\oauth2_client\credentials_manager.py", line 87, in _handle_bad_response
    raise OAuthError(HTTPStatus(response.status_code), error.get('error'), error.get('error_description'))
oauth2_client.credentials_manager.OAuthError: 400  - server_error : Internal Server Error
2024-02-07 11:27:26,398 :: INFO :: 127.0.0.1 - - [07/Feb/2024 11:27:26] "POST /_dash-update-component HTTP/1.1" 200 -

Reply:
Looks like you're testing old code there. This self.manager.init_with_user_credentials_realm from your error message has been renamed in https://github.com/flobz/psa_car_controller/pull/754/commits/66c188f18cee5270da90c0dfbf42bc6567128a48

Reply:
Yes you are right.
Now I get peugeot link, but it still shows this error while trying to access 
{ "error": "redirect_uri_mismatch", "error_description": "The redirection URI provided does not match a pre-registered value." }

Reply:
Oha. The "op" in "mym**op**" stands for Opel then I guess 😅 🤦
So let's pass the brand_code into the URL too...

Reply:
> Oha. The "op" in "mym**op**" stands for Opel then I guess 😅 🤦 So let's pass the brand_code into the URL too...

This worked for me, Thanks :-)

However, I have no idea how to incorporate this in to HA